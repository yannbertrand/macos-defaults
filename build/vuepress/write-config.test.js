const fs = require('fs')
jest.mock('fs')

const writeConfig = require('./write-config')

const templatesPath = 'templates'
const destinationPath = 'dist'
const copiedFiles = ['.vuepress/public/favicon.ico']

describe('write-config', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  beforeEach(() => {
    callWriteConfig()
  })

  it('should copy some static files', () => {
    copiedFiles.forEach((file) => {
      const fileContent = readFile(`${destinationPath}/${file}`)
      expect(fileContent).toEqual(`copied:${templatesPath}/${file}`)
    })
  })

  it('should write a vuepress config.yml file using the template', () => {
    const vuepressConfigContent = readFile(
      `${destinationPath}/.vuepress/config.yml`
    )
    expect(vuepressConfigContent).toMatchSnapshot()
  })
})

const supportedLanguages = {
  languages: [
    {
      url: '/',
      lang: 'en-US',
      home: 'Home',
      defaults: {
        categories: [
          {
            folder: 'category1',
            name: 'Category 1',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
          {
            folder: 'category2',
            name: 'Category 2',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
        ],
      },
    },
    {
      url: '/fr/',
      lang: 'fr-FR',
      home: 'Accueil',
      defaults: {
        categories: [
          {
            folder: 'categorie1',
            name: 'Catégorie 1',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
          {
            folder: 'categorie2',
            name: 'Catégorie 2',
            keys: [
              {
                key: 'page',
                title: 'Page',
              },
            ],
          },
        ],
      },
    },
  ],
}

const callWriteConfig = () =>
  writeConfig(supportedLanguages, templatesPath, destinationPath)
const readFile = (file) => fs.readFakeFileSync(file, 'utf8')
