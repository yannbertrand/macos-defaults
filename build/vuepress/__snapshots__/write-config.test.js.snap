// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`write-config should write a vuepress config.yml file using the template 1`] = `
"---

title: 'macOS defaults'
description: 'Uncomplete list of macOS defaults commands with demos ✨'
evergreen: true
plugins:
  - '@vuepress/medium-zoom'
locales:
  /:
    lang: 'en-US'
  /fr/:
    lang: 'fr-FR'
themeConfig:
  repo: 'yannbertrand/macos-defaults'
  locales:
    /:
      nav:
        - text: 'Home'
          link: '/'
      activeHeaderLinks: false
      sidebar:
        - title: 'Home'
          path: '/'
        - title: Category 1
          collapsable: false
          path: '/category1/'
          children:
            - '/category1/page'
        - title: Category 2
          collapsable: false
          path: '/category2/'
          children:
            - '/category2/page'
    /fr/:
      nav:
        - text: 'Accueil'
          link: '/fr/'
      activeHeaderLinks: false
      sidebar:
        - title: 'Accueil'
          path: '/fr/'
        - title: Catégorie 1
          collapsable: false
          path: '/fr/categorie1/'
          children:
            - '/fr/categorie1/page'
        - title: Catégorie 2
          collapsable: false
          path: '/fr/categorie2/'
          children:
            - '/fr/categorie2/page'
"
`;
